python/graphstorm/sagemaker/sagemaker_gb_convert.py

```diff
@@ -0,0 +1,95 @@
+""" Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SageMaker implementation of DistDGL to GraphBolt conversion.
+"""
+
+import importlib.metadata
+import logging
+import os
+import time
+from packaging import version
+
+import boto3
+import sagemaker
+from sagemaker.s3 import S3Uploader
+
+from graphstorm.gpartition.convert_to_graphbolt import run_gb_conversion
+
+
+def run_gb_convert(s3_output_path: str, local_dist_part_config: str, njobs: int):
+    """Convert DistDGL partitions to GraphBolt format on SageMaker Processing.
+
+
+    Parameters
+    ----------
+    s3_output_path : str
+        The S3 prefix under which we will upload the GB converted graph files.
+    local_dist_part_config : str
+        The local path to the downloaded partition data metadata.json file.
+    njobs: int
+        Number of parallel processes to use during GraphBolt conversion.
+        One process per partition is used, so ``njobs`` partitions needs
+        to be able to fit into memory.
+
+    Raises
+    ------
+    ValueError
+        If the version of DGL used is under 2.1.0
+    """
+    assert os.path.isfile(
+        local_dist_part_config
+    ), f"{local_dist_part_config=} does not exist."
+
+    dgl_version = importlib.metadata.version("dgl")
+    if version.parse(dgl_version) < version.parse("2.1.0"):
+        raise ValueError(
+            "GraphBolt conversion requires DGL version >= 2.1.0, "
+            f"but DGL version was {dgl_version}. "
+        )
+
+    boto_session = boto3.Session(region_name=os.environ['AWS_REGION'])
```

Comment:
Do we have boto3 dependencies in our docker?

Reply:
`sagemaker` depends on `boto` and `sagemaker` is installed on all SM images we use so we should be OK here.

I am planning a small update to our SM images to pin sagemaker, boto and botocore to specific versions.